chainstate/src/detail/pow/work.rs

```diff
@@ -74,7 +74,8 @@ fn calculate_work_required<H: BlockIndexHandle>(
         PoWStatus::Ongoing => {
             let prev_block_id = header
                 .prev_block_id()
-                .clone()
+                .classify(chain_config)
+                .block_id()
                 .expect("If PoWStatus is `Onging` then we cannot be at genesis");
```

Comment:
I think this is one of the examples we'd like to avoid. The better way of dealing with this, IMHO, is to just get the id, as is, and use it as if it's definitly not genesis. If it's not genesis, it won't be found. Hence, never causing a crash. The other day Aaro bumped, by coincidence, by a case where such an expect triggers if the wrong block is submitted.

Reply:
I am not sure I understand the comment completely. Let me just point out that this checks parent's ID and whether it is genesis, not whether the current block is a genesis. This kind of reasoning about block's parents/ancestors, while desirable, is out of scope of the present PR.

Similar issue comes up in orphan pool: A block with genesis as a parent (transitively) has no business being in orphan pool, so some `expect`s remain there. That should be addressed separately, although it would be nice to have too.

Reply:
> Let me just point out that this checks parent's ID and whether it is genesis

A peer can set the parent id to whatever they want. That's the issue.

If it's in the orphan's pool, that's not good, but remember that nothing goes into the orphan's pool unless it's from local source.

Reply:
I'm resolving this as it's not a new issue introduced in this PR. The expect was there before, and I think the resolution is out of scope for the present PR.